### PR TITLE
fix(graphql): readonly arrays not allowed in decorator input

### PR DESCRIPTION
**Pull request by CarsonF**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

```ts

const interfaces = [A, B] as const;

@ObjectType({
  implements: interfaces, // error
})
```

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/graphql/lib/utils/types.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export type Many<T> = T | readonly T[];
```
Comment (micalevisk) [inline packages/graphql/lib/utils/types.ts:1]:
```suggestion
export type Many<T> = T | ReadonlyArray<T>;
```

this should work. See https://github.com/nestjs/nest/pull/13655

File: packages/graphql/lib/decorators/resolve-field.decorator.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ export type ResolveFieldOptions<T = any> = BaseTypeOptions<T> & {
   /**
    * Array of middleware to apply.
    */
-  middleware?: FieldMiddleware[];
+  middleware?: readonly FieldMiddleware[];
```
Comment (micalevisk) [inline packages/graphql/lib/decorators/resolve-field.decorator.ts:42]:
```suggestion
  middleware?: ReadonlyArray<FieldMiddleware>;
```

**Review by micalevisk — CHANGES_REQUESTED** [review]
minor changes to support both array and read only arrays

**Comment by CarsonF** [comment]
> minor changes to support both array and read only arrays

These represent the same thing. Mutable arrays are a superset of read-only ones. But if this style is preferred I can switch it.

**Comment by micalevisk** [comment]
I just noticed something that would break due to that `ReadonlyArray`:

```ts
const imports: ResolveFieldOptions['middleware'] = [];
imports.push(() => {})
```

because we can't invoke `.push` on readonly arrays

and I believe that we shouldn't prevent that usage

**Pull request closed without merging**